### PR TITLE
[WIP] android jcenter fix 

### PR DESCRIPTION
**Pull request by daniellacosse**
Jcenter/bintray is broken, but our `cordova-android` fetches from there. Fix for this [was introduced July of last year](https://github.com/apache/cordova-android/pull/1251).

- Upgrades `cordova-android` to latest.
- TODO: Address breaking change introduced to `WebViewAssetLoader` in `v10.0.0` (https://cordova.apache.org/announcements/2021/07/20/cordova-android-10.0.0.html)

### CODE REVIEW

**Review by fortuna — COMMENTED** [review]
The change looks fine, but we need to confirm if storage if preserved after an upgrade.

**Comment by daniellacosse** [comment]
> The change looks fine, but we need to confirm if storage if preserved after an upgrade.

I presume setting `<preference name="AndroidInsecureFileModeEnabled" value="true" />` isn't a viable option?

Is everything exclusively stored via local storage? I haven't tested it yet, but if so it's very likely this will dump everyone's configs.

**Comment by daniellacosse** [comment]
> > The change looks fine, but we need to confirm if storage if preserved after an upgrade.
> 
> I presume setting `<preference name="AndroidInsecureFileModeEnabled" value="true" />` isn't a viable option?
> 
> Is everything exclusively stored via local storage? I haven't tested it yet, but if so it's very likely this will dump everyone's configs.

seems mostly like it's the language of choice, unless vscode is hiding results from me again

https://github.com/Jigsaw-Code/outline-client/blob/0571eafe2d14a33b2e3bb8c329169fefcc2805bb/src/www/app/app.ts#L249-L250

**Comment by fortuna** [comment]
We use localStorage for everything: https://github.com/Jigsaw-Code/outline-client/search?q=localStorage&type=

We had to migrate uses on iOS from file:// to app://localhost once and it was a pain. You'll see the code there.

I think the `AndroidInsecureFileModeEnabled` is ok, since we don't load files from file:// that we don't trust.
I'm ok with setting that, but I want to make sure the data is preserved in that case too. And if we are testing that the data is preserved, we might asa well test without that option too.

**Review by fortuna — COMMENTED** [review]
Let me know when you have verified it's working with the update.

**Comment by daniellacosse** [comment]
> We use localStorage for everything: https://github.com/Jigsaw-Code/outline-client/search?q=localStorage&type=
> 
> We had to migrate uses on iOS from file:// to app://localhost once and it was a pain. You'll see the code there.
> 
> I think the `AndroidInsecureFileModeEnabled` is ok, since we don't load files from file:// that we don't trust. I'm ok with setting that, but I want to make sure the data is preserved in that case too. And if we are testing that the data is preserved, we might asa well test without that option too.

In the search you linked, the only call to `setItem` was to the `overrideLanguage` key, so you can understand my confusion. I assume then that some third-party is using it somehow.

Apologies on the delay, I'm trying to figure out how best to do this.

File: .java-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.8
```
Comment (daniellacosse) [inline .java-version:1]: This was useful while trying to get the app to build on my local machine. `jenv` uses it to manage which version of java it should use - our app uses v1.8 while android studio with blaze defaults to 11.0.11

**Review by daniellacosse — COMMENTED** [review]
ah, I was being a bit silly - I never got the android app to build on my local mac and was trying to see if I could solve that, either with `Android Studio w/ Blaze` (recency bias 😅) or manually - only to realize I _can't_ build the old version locally because of this very issue!!

Then, of course, I remembered that the build system has been building versions of the android app this whole time 🤦 . Here are the builds of note while I struggle to figure out what how to install each apk on my emulator:

- [https://drive.google.com/file/d/18wIZ55Xw_Ix6SgmS3ekW64hpIELgce_p/view?usp=sharing](cordova-android v9 build)
- [https://drive.google.com/file/d/193QCkF6iT3FgVjvCRCgUc5jnm3fxRrJw/view?usp=sharing](cordova-android v10 build)
- [https://drive.google.com/file/d/193zGxl1xoyP6VAE-LSiR8MKah8kRhzPa/view?usp=sharing]("unsafe file" cordova-android v10 build)

My main concern right now is - will the emulator preserve `localStorage` across the different apks? Or does it treat each as its own separate app? I don't even know how that works.

File: .java-version (hunk 1 above)
Comment (jyyi1) [reply .java-version:1]: Maybe a question for @fortuna , shall we upgrade Java to a newer version considering we are approaching the end-of-support time of Java 8: https://endoflife.date/java.

**Review by fortuna — APPROVED** [review]
Let's submit this as is to unblock the other stuff, with the understanding that we still need to test it.

File: .java-version (hunk 1 above)
Comment (fortuna) [reply .java-version:1]: Yes, let's upgrade to the newest Java if we can. But in another PR

**Pull request merged**